Add unpaginated versions of multi-row get methods (w/tests)

slaarti opened the pull request:
## Description

Per #59, this PR is to add new methods to `FastCRUD` to do multi-row gets (without and with joins) that return all rows in a single list without pagination.

## Changes

Added new methods `get_multi_all` and `get_multi_joined_all` that are basically copies minus the `offset`, `limit`, and `return_total_count` args as well as without wrapping the list of result rows in a `dict` (since that happens in the existing methods to help with pagination).

## Tests

`test_get_multi.py` and `test_get_multi_joined.py` for both `sqlalchemy` and `sqlmodel` were copied into new files and adjusted to test the new methods (including removing tests that only made sense in the context of pagination).

## Checklist
- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [X] I have added necessary documentation (if appropriate).
- [X] I have added tests that cover my changes (if applicable).
- [X] All new and existing tests passed.

igorbenav commented:
@slaarti could you please refactor? I think doing it [this way](https://github.com/igorbenav/fastcrud/issues/59#issuecomment-2076287486) would be bettet indeed

slaarti commented:
Okay, sure. Not tonight, might take a couple days, but I'm fine with the alternate approach. 👍

igorbenav commented:
Nice, thanks!

slaarti commented:
Okay, I rolled back the original branch (well, renamed it and did a fresh checkout) and implemented the alternate method, including checks on the endpoint creator and crud router (with a flag to bypass the checks). The force push is for the new branch.

slaarti commented:
I'd opted for `0` as much to avoid having to deal with changes in typing and looking for `None` vs. `0` as feeling like saying you want zero rows made no sense... but yeah, okay, you make a good point. Will do. 👍

slaarti commented:
...Actually, come to think of it, an interesting aspect of changing it to `None` is that I'm pretty sure there's no good way to pass `None` as a path/query parameter, which eliminates the need for the changes made for sanitizing the automatically generated endpoints.

Anyone who wants to provide an unlimited get in their (presumably private, as in my cases) API will have to create an endpoint specifically for it, but I'm okay with that.

igorbenav commented:
I guess everything worked out even better, great!
Thanks @slaarti and welcome to fastcrud contributors🎉
